catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/MetacardCreator.java

```diff
@@ -0,0 +1,110 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.common.tika;
+
+import java.io.Serializable;
+import java.util.Date;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.metadata.TikaCoreProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.data.impl.MetacardImpl;
+
+/**
+ * Creates {@link Metacard}s from Tika {@link Metadata} objects.
+ */
+public class MetacardCreator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetacardCreator.class);
+
+    /**
+     * A convenience method for creating a new {@link Metacard} of type
+     * {@link BasicTypes#BASIC_METACARD} from a {@link Metadata} object.
+     *
+     * @param metadata    the {@code Metadata} object containing the metadata relevant to the
+     *                    metacard, must not be null
+     * @param uri         resource URI for the metacard, may be null
+     * @param metadataXml metadata XML to place in the metacard's metadata field, may be null
+     * @return a new {@code Metacard}
+     */
+    public static Metacard createBasicMetacard(final Metadata metadata, final String uri,
+            final String metadataXml) {
+        final Metacard metacard = new MetacardImpl(BasicTypes.BASIC_METACARD);
+
+        final String contentType = metadata.get(Metadata.CONTENT_TYPE);
+        if (StringUtils.isNotBlank(contentType)) {
+            metacard.setAttribute(new AttributeImpl(Metacard.CONTENT_TYPE, contentType));
+        }
+
+        final String title = metadata.get(TikaCoreProperties.TITLE);
+        if (StringUtils.isNotBlank(title)) {
+            metacard.setAttribute(new AttributeImpl(Metacard.TITLE, title));
+        }
+
+        final String createdDateStr = metadata.get(TikaCoreProperties.CREATED);
+        final Date createdDate = convertDate(createdDateStr);
+        if (createdDate != null) {
+            metacard.setAttribute(new AttributeImpl(Metacard.CREATED, createdDate));
+        }
+
+        final String modifiedDateStr = metadata.get(TikaCoreProperties.MODIFIED);
+        final Date modifiedDate = convertDate(modifiedDateStr);
+        if (modifiedDate != null) {
+            metacard.setAttribute(new AttributeImpl(Metacard.MODIFIED, modifiedDate));
+        }
+
+        if (StringUtils.isNotBlank(uri)) {
```

Comment:
Oops, this actually isn't a URI. It's supposed to be an ID, but for some reason `TikaInputTransformer` treats it as a URI.